charts/consul/templates/gossip-encryption-autogeneration-job.yaml

```diff
@@ -0,0 +1,76 @@
+{{- if .Values.global.gossipEncryption.autoGenerate }}
+{{- if (or .Values.global.gossipEncryption.secretName .Values.global.gossipEncryption.secretKey) }}
+  {{ fail "If global.gossipEncryption.autoGenerate is true, global.gossipEncryption.secretName and global.gossipEncryption.secretKey must not be set." }}
+{{ end }}
+---
+# automatically generate encryption key for gossip protocol and save it in Kubernetes secret
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "consul.fullname" . }}-gossip-encryption-autogenerate
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded,before-hook-creation
+spec:
+  template:
+    metadata:
+      name: {{ template "consul.fullname" . }}-gossip-encryption-autogenerate
+      labels:
+        app: {{ template "consul.name" . }}
+        chart: {{ template "consul.chart" . }}
+        release: {{ .Release.Name }}
+        component: gossip-encryption-autogeneneration
+      annotations:
+        "consul.hashicorp.com/connect-inject": "false"
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ template "consul.fullname" . }}-gossip-encryption-autogenerate
+      containers:
+        - name: gossip-encryption-autogen
```

Comment:
I forgot to mention that we should add [security context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) to this pod so that it doesn't run as root. We're using the consul image here, and consul docker image runs as root by default unless you overwrite it. Here is an example of how we do it for the consul server:

https://github.com/hashicorp/consul-k8s/blob/9a974c1a9c9fe0594460352b25ef05fd4ba13734/charts/consul/templates/server-statefulset.yaml#L79

https://github.com/hashicorp/consul-k8s/blob/9a974c1a9c9fe0594460352b25ef05fd4ba13734/charts/consul/values.yaml#L508-L512

Reply:
Yes, and I'm going to just hardcode the security context because there is no reason this should ever need to run as root.

Reply:
Done!